### PR TITLE
fix(26353): fix flaky test Request-queue UI changes should show network specific to domain @no-mmi

### LINKED ISSUE #26353: Fix "Request-queue UI changes should show network specifi..." flaky tests
https://app.circleci.com/pipelines/github/MetaMask/metamask-extension/95075/workflows/9d5509ee-8355-4264-92da-c08e711b04b9/jobs/3538478/tests

### PR DESCRIPTION
**Pull request by DDDDDanica**
## **Description**
After investigation, it is not very clear that the reject transaction is finished before we switch to the second dapp to perform send transaction action. Hence we put an await for transaction to be finished. 

This test is ran locally 20 times with no failure, and the report is attached here: 
[command_output.txt](https://github.com/user-attachments/files/16551755/command_output.txt)

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/26354?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/26353

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: test/e2e/tests/request-queuing/ui.spec.js [hunk 1]
```diff
@@ -194,9 +194,6 @@ describe('Request-queue UI changes', function () {
       async ({ driver }) => {
         await unlockWallet(driver);
 
-        // Navigate to extension home screen
```
Comment (DDDDDanica) [inline test/e2e/tests/request-queuing/ui.spec.js:197]: We are on extension home page after unlock so this step is unnecessary.

File: test/e2e/tests/request-queuing/ui.spec.js [hunk 2]
```diff
@@ -213,22 +210,24 @@ describe('Request-queue UI changes', function () {
...
           chainId: '0x539',
           networkText: 'Localhost 8545',
           originText: DAPP_URL,
         });
         await rejectTransaction(driver);
+        await driver.waitUntilXWindowHandles(3);
 
         // Go to the second dapp, ensure it uses Ethereum Mainnet
-        await selectDappClickSend(driver, DAPP_ONE_URL);
+        await switchToDappAndSendTransaction(driver, DAPP_ONE_URL);
```
Comment (DDDDDanica) [inline test/e2e/tests/request-queuing/ui.spec.js:223]: rename to illustrate the function better

**Review by desi — APPROVED** [review]
LGTM

**Review by hjetpoluru — APPROVED** [review]
LGTM

**Review by Gudahtt — CHANGES_REQUESTED** [review]
The `should autoswitch networks to the last used network for domain` test in this file is consistently failing since the last update with `develop`, caused by a timeout. The test stalls and times out after 20 minutes.

**Comment by DDDDDanica** [comment]
Hey @Gudahtt Can you show me the pipeline where that test is failing? I checked the ci and the flakiness are from other tests , running this test locally didn't not result in a failure

**Comment by sunobun** [comment]
Support

**Comment by Gudahtt** [comment]
Check out the logs for the `test-e2e-chrome-webpack` job. You can see that it runs for a very long time, and the last output is for the test run "Request-queue UI changes should signal from UI to dapp the network change @no-mmi". And it exits with a code of 124, indicating a timeout. The most recent three failing runs failed this exact same way.

It seems that due to the timeout, the test was not registered as a failing test, so it was skipped when "rerun failed tests" was run 🤔 . That's concerning, I didn't realize that "rerun failed tests" could skip tests from being run at all.

**Comment by DDDDDanica** [comment]
close due to the fix from https://github.com/MetaMask/metamask-extension/pull/26449

**Pull request closed without merging**